vision/cloud-client/product_search/product_in_product_set_management.py

```diff
@@ -117,6 +119,45 @@ def remove_product_from_product_set(
 # [END vision_product_search_remove_product_from_product_set]
 
 
+# [START vision_product_search_purge_products_in_product_set]
+def purge_products_in_product_set(
+        project_id, location, product_set_id, force):
+    """Delete all products in a product set.
+    Args:
+        project_id: Id of the project.
+        location: A compute region name.
+        product_set_id: Id of the product set.
+        force: Perform the purge only when force is set to True.
+    """
+    client = vision.ProductSearchClient()
+
+    parent = client.location_path(
+        project=project_id, location=location)
+
+    product_set_purge_config = vision.types.ProductSetPurgeConfig(
+        product_set_id=product_set_id)
+
+    # The purge operation is async.
+    operation = client.purge_products(
+        parent=parent,
+        product_set_purge_config=product_set_purge_config,
+        # The operation is irreversible and removes multiple products.
+        # The user is required to pass in force=True to actually perform the
+        # purge.
+        # If force is not set to True, the service raises an exception.
+        force=force)
+
+    # TODO: replace with operation.result(timeout=60)

```

Comment:
Adding a GitHub PR comment for this TODO to track it ~ this comment will disappear once the TODO is fixed :D